File: hh-front/src/models/categoria_viaje_model.js
import { UsuarioModel } from './usuario_model'

export class CategoriaViajeModel {
  constructor () {
    this.id = ''
    this.categoria = ''

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}

File: hh-back/src/main/java/gloit/hiperionida/helios/model/CategoriaViajeModel.java
package gloit.hiperionida.helios.model;

import gloit.hiperionida.helios.util.model.AbsAuditoriaModel;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "categoria_viaje")
public class CategoriaViajeModel extends AbsAuditoriaModel {
    private String categoria;
    @Column(columnDefinition = "TEXT")
    private String notas;

    public CategoriaViajeModel() {
    }

    public CategoriaViajeModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, String categoria, String notas) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada);
        this.categoria = categoria;
        this.notas = notas;
    }
}

